File: src/proxy/examples/owneranonymizer/AnonymousOwnerProxy.java
package proxy.examples.owneranonymizer;

/**
 * RNF1.2: keep owner personal data anonymous for anonymous sighting reporters.
 * Proxy hides real contact details behind a public alias.
 */
public class AnonymousOwnerProxy implements OwnerContact {

    private final RealOwnerContact real;
    private final String alias;

    public AnonymousOwnerProxy(RealOwnerContact real, String alias) {
        this.real = real;
        this.alias = alias;
    }

    @Override public String getName()  { return alias; }
    @Override public String getPhone() { return "[oculto]"; }
    @Override public String getEmail() { return "[oculto]"; }

    /** Only authenticated flow may unwrap (not exposed to public API). */
    RealOwnerContact unwrap() { return real; }
}

File: src/proxy/examples/owneranonymizer/AnonymousOwnerProxy.js
'use strict';

const OwnerContact = require('./OwnerContact');

/**
 * RNF1.2: datos personales del dueno permanecen anonimos para ciudadanos
 * que reportan avistamientos. Proxy oculta contacto real detras de alias.
 */
class AnonymousOwnerProxy extends OwnerContact {
  constructor(real, alias) {
    super();
    this._real = real;
    this.alias = alias;
  }
  getName()  { return this.alias; }
  getPhone() { return '[oculto]'; }
  getEmail() { return '[oculto]'; }

  _unwrap() { return this._real; }
}

module.exports = AnonymousOwnerProxy;
